include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -669,6 +669,18 @@ get(const oneapi::dpl::__internal::tuple<_Tp...>&& __a)
 {
     return ::std::move(__a).template get<_Idx>();
 }
+
+// To enable oneapi::dpl::zip_iterator to satisfy the requirements for the std::input_iterator concept in C++20,
+// we must provide a basic_common_reference specialization for our internal tuple implementation.
+#if _ONEDPL_CPP20_CONCEPTS_PRESENT
+template <typename... TTypes, typename... UTypes, template <typename> typename TQual,
+          template <typename> typename UQual>
+struct basic_common_reference<oneapi::dpl::__internal::tuple<TTypes...>, oneapi::dpl::__internal::tuple<UTypes...>,
+                              TQual, UQual>
```

Comment:
I'm having trouble understanding `TQual` and `UQual`.  

Are they just something which is required for this function for it to be of use to the Standard library implementation? 
Perhaps it uses these to add type qualifiers like `const` to the types?

Its not very clear to me as it is written in the standard.

Reply:
They are required for this function to be used by the standard library implementation. My understanding is that they add cvref-qualifiers to the types.

It is mentioned briefly in the explanation of [std::common_reference's](https://en.cppreference.com/w/cpp/types/common_reference) required behavior:

"Otherwise, if `std::basic_common_reference<std::remove_cvref_t<T1>, std::remove_cvref_t<T2>, T1Q, T2Q>::type` exists, where TiQ is a unary alias template such that TiQ<U> is U with the addition of Ti's cv- and reference qualifiers, then the member type type names that type;"